src/DNSC/Server.hs

```diff
@@ -41,15 +41,17 @@ udpSockets port = mapM aiSocket . filter ((== Datagram) . addrSocketType) <=< ad
 run :: Log.FOutput -> Log.Level -> Int -> Bool -> Int
     -> PortNumber -> [HostName] -> Bool -> IO ()
 run logOutput logLevel maxCacheSize disableV6NS conc port hosts stdConsole = do
-  monParams <- bind logOutput logLevel maxCacheSize disableV6NS conc port hosts
+  (serverLoops, monParams) <- bind logOutput logLevel maxCacheSize disableV6NS conc port hosts
   monLoops <- uncurry (uncurry $ uncurry $ monitor stdConsole) monParams
-  foldr concurrently_ (return ()) monLoops
+  race_
+    (foldr concurrently_ (return ()) serverLoops)
+    (foldr concurrently_ (return ()) monLoops)
 
 type QSizeInfo = (IO (Int, Int), IO (Int, Int), IO (Int, Int), IO (Int, Int))
 
 bind :: Log.FOutput -> Log.Level -> Int -> Bool -> Int
```

Comment:
Now this function does a lot of jobs. So, `bind` is misleading. What about `setup`?

Reply:
https://github.com/khibino/dns-resolver/pull/19/commits/96f4074b1be572f05540fe0a7b61f93e80392498